### PR TITLE
feat: add send payment pending UI

### PR DESCRIPTION
**Pull request by ovitrif**
This PR adds a pending payment UI for Lightning sends that take longer than expected (e.g. hold invoices).

### Description

1. Adds a `SendPendingScreen` with an hourglass animation, shown when a Lightning payment exceeds the 10-second timeout without resolving
2. Adds `PendingPaymentResolution` tracking in `LightningRepo` so payment success/failure events are routed back to the pending screen
4. Handles both resolution paths: navigates to success sheet on completion, or error sheet on failure
5. Handles invoices in background if sheet was dismissed with toast for success or error
6. Adds a "View Details" button that links to the activity detail once the payment activity is synced
7. Toasts for the pending payment handled by the sheet only fire when the user has closed the sheet, if pending sheet is visible, rely on sheets UI.
8. Adds mechanism to ensure older pending payments still show toast on settled/canceled, even if UI shows sheet for new pending payment.

#### Recording
- **Note**: Video speed is at 1.75x playback speed.
-  Script
    1. Pending UI > Details > List > Settle > Error Toast
    2. Pending UI > Cancel > Error Sheet
    3. Pending UI > Settle > Success Sheet
    4. Pending UI > Details > List > Cancel Toast

| Recording | Edge Cases |
| - | - |
| <video src="https://github.com/user-attachments/assets/97acdad6-d317-4502-bbc6-b72db3fe2b0d"> | <video src="https://github.com/user-attachments/assets/8965eaed-2fd1-4517-82d8-3528da3f783c"> |

#### Setup
1. Startup `bitkit-docker`: `docker compose up -d`
2. Start emulator via Android Studio
5. Run this first:
    ```sh
    adb reverse tcp:9735 tcp:9735 && \
    adb reverse tcp:3000 tcp:3000 && \
    adb reverse tcp:60001 tcp:60001 && adb reverse --list
    ```
6. Apply [local-electrum.patch](https://gist.github.com/ovitrif/ac13a7ba09efd4c12bee0a1b4cc27f4a) to connect to local Electrum, then build and run on emulator:
    ```sh
    curl -sL https://gist.githubusercontent.com/ovitrif/ac13a7ba09efd4c12bee0a1b4cc27f4a/raw/local-electrum.patch | git apply
    ```
9. [Open a channel](https://github.com/synonymdev/bitkit-docker/blob/main/README.md#external-node-channel), skip first 4 steps

Use `./bitcoin-cli` in `bitkit-docker` for hold invoices ops:

```sh
./bitcoin-cli holdinvoice 42 -m 'test' # create
./bitcoin-cli settleinvoice __preimage__ # settle
./bitcoin-cli cancelinvoice __hash__ # cancel
```

#### Tests
- [x] Pending screen should show on hold invoices
- [x] Payment resolves to success sheet when on sheet
- [x] Payment resolves to error sheet when on sheet
- [x] View Details button opens pending activity detail
- [x] Close button + settle/cancel -> resolves to toast
- [x] UI should match FIGMA / iOS
- [x] QuickPay for cancelled/settled pending payment
- [x] Tests from stacked PR #828
  - [x] App in BG + settle/cancel → success/error notification

#### Edge Cases
No need for reviewer(s) to test edge cases, they're listed here for transparency into what was dev-tested.
- [x] Back nav from success/error sheet should dismiss it
- [x] Pending screen visible + settle → no toast, success screen shown
- [x] Pending screen visible + cancel → no toast, error screen shown
- [x] Sheet closed + settle → toast shown
- [x] Sheet closed + cancel → toast shown
- [x] Multi pending payment resolution
  - Pay hold invoice1 (old) + dismiss sheet
  - Pay hold invoice2 (new) + settle/cancel old invoice while on pending sheet → toast shown
  - Settle/cancel new invoice → no toast, success/error screen shown
- [x] Scanner (QuickPay on/off), paste, cancel invoice, tap 'Try Again' → Send Recipient shown
- [x] Paste 0-amount hold invoice via send + cancel + try again → Send Recipient

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/java/to/bitkit/viewmodels/QuickPayViewModel.kt [hunk 1]
```diff
@@ -62,6 +63,18 @@ class QuickPayViewModel @Inject constructor(
                         )
                     }
                 }.onFailure { error ->
+                    if (error is PaymentPendingException) {
+                        Logger.info("QuickPay lightning payment pending: ${error.paymentHash}")
```
[the remark this answers is not in the document]
Comment (ovitrif) [reply app/src/main/java/to/bitkit/viewmodels/QuickPayViewModel.kt]: fixed in 8215e5b9

**Review by jvsena42 — COMMENTED** [review]
Code looks good overall, will test in #828
